### DISCUSSION TITLE
Support pnpm's lockfile when deploying

### DISCUSSION THREAD

**Discussion by redstrike** [vercel/vercel#5559] [Ideas] [answered]
I'm using PNPM and everything is great until I see a little problem while deploying to Vercel.
It seems Vercel's build process doesn't support **`pnpm-lock.yaml`**, so this info message always showing up.
```
info No lockfile found.
```
I wonder if Vercel can support **PNPM's lock file**?

**Comment by cyrus-za** [comment]
@redstrike use a custom [install command](https://vercel.com/docs/build-step#install-command) in your dashboard to do `npx pnpm i`

**Reply by redstrike** [reply]
@cyrus-za thank you very much! I forgot that Vercel supports custom install command.

**Reply by redstrike** [reply]
@cyrus-za this is my follow up. It doesn't work as expected. Vercel detects `no-lockfile` and will try to use `yarn install` instead.

**Reply by cyrus-za** [reply]
do you not commit your lockfile?

**Reply by redstrike** [reply]
@cyrus-za yes, we have to commit it? I see the lockfile has been uploaded to Vercel, is that not enough?

**Comment by styfle** [comment, marked as answer]
pnpm is supported without any additional configuration since March 2022

https://vercel.com/changelog/projects-using-pnpm-can-now-be-deployed-with-zero-configuration

You can also select a version of pnpm with Corepack since July 2022

https://vercel.com/changelog/corepack-experimental-is-now-available
